libminifi/test/unit/FileUtilsTests.cpp

```diff
@@ -52,43 +52,43 @@ TEST_CASE("TestFileUtils::concat_path", "[TestConcatPath]") {
 
 TEST_CASE("TestFileUtils::get_parent_path", "[TestGetParentPath]") {
 #ifdef WIN32
-  REQUIRE("foo\\" == FileUtils::get_parent_path("foo\\bar"));
-  REQUIRE("foo\\" == FileUtils::get_parent_path("foo\\bar\\"));
-  REQUIRE("C:\\foo\\" == FileUtils::get_parent_path("C:\\foo\\bar"));
-  REQUIRE("C:\\foo\\" == FileUtils::get_parent_path("C:\\foo\\bar\\"));
-  REQUIRE("C:\\" == FileUtils::get_parent_path("C:\\foo"));
-  REQUIRE("C:\\" == FileUtils::get_parent_path("C:\\foo\\"));
-  REQUIRE("" == FileUtils::get_parent_path("C:\\"));  // NOLINT(readability-container-size-empty)
-  REQUIRE("" == FileUtils::get_parent_path("C:\\\\"));  // NOLINT(readability-container-size-empty)
+  CHECK("foo" == FileUtils::get_parent_path("foo\\bar"));
+  CHECK("foo\\bar" == FileUtils::get_parent_path("foo\\bar\\"));
```

Comment:
I think this is a bit confusing that both refer to the `foo\\bar` directory but they have a different result when calling `get_parent_path`. get_child_path has a similar problem. Either the functionality should be changed or the current functionality should be better documented (in comments) and the better represented in the name of the functions.

Reply:
This is how `std::filesystem::path::parent_path()` and `std::filesystem::path::filename()` work: `std::filesystem::path{"foo\\bar\\"}` (or `std::filesystem::path{"foo/bar/"}` on Linux) do not refer to the `bar` directory, but to a file with an empty filename in the `bar` directory.

I agree this is unexpected, but as this is how the standard library works, I figured I'd change the code where we call it instead of wrapping the standard library functions to change their behavior.

Reply:
I see, if we don't want to deviate from the semantics of the standard library, could you add this explanation as a comment to these functions?

Reply:
I have added some comments in a432d9485bee1f0b6d858ddb31dcf569cc85e376